mono/metadata/class-init.c

```diff
@@ -3613,6 +3648,8 @@ mono_class_layout_fields (MonoClass *klass, int base_instance_size, int packing_
 		if (blittable) {
 			if (field->type->byref || MONO_TYPE_IS_REFERENCE (field->type)) {
 				blittable = FALSE;
+			} else if (mono_type_is_generic_parameter (field->type) && mono_class_is_gparam_with_reference_parent (mono_class_from_mono_type_internal (field->type))) {
+				blittable = FALSE;
```

Comment:
Would it hurt things if we would set blittable=FALSE for all gparams ?

Reply:
Yes (https://github.com/mono/mono/pull/15761#issuecomment-513379827)

Reply:
We're not looking at whether a gparam like `T` is blittable here.  Rather, we're inside a GTD like `G<T>` that has some field `T field` and we're trying to decide whether that makes `G<>` blittable or not.

What I wonder is whether we actually care about GTDs like `G<>` or open instantiations like `G<List<T>>` being blittable, or if we only care about closed instantiations.

And I think we the answer is we do care, because of generic sharing.  Something like `G<List<T_REF>>` needs to have a valid `klass->blittable` value.

Reply:
Note that the gshared type parameters have this encoded in param->gshared_constraint.